### ISSUE TITLE
`make generate-doc` doesn't update CRD API doc when using Go 1.16.x

### ISSUE THREAD

**Issue by tadayosi** [apache/camel-k#3236]
When running `make generate-doc` target, it shows the following warnings `can only use path@version syntax with 'go get' and 'go install' in module-aware mode` and although it says the generation done it does not update the CRD API doc.
```
$ make generate-doc 
./script/gen_doc.sh
Generating API documentation...
Generating CRD API documentation...
package github.com/djencks/gen-crd-api-reference-docs@e63530f10b55be5f2d82e223d83f86c13e5158e5: can only use path@version syntax with 'go get' and 'go install' in module-aware mode
package github.com/djencks/gen-crd-api-reference-docs@e63530f10b55be5f2d82e223d83f86c13e5158e5: can only use path@version syntax with 'go get' and 'go install' in module-aware mode
Generating CRD API documentation... Done.
Generating API documentation... done!
```

The problem is that `go run path@version` used in `./script/gen_crd/gen_crd_api.sh` is really not allowed.

Go version:
```
$ go version 
go version go1.16.15 linux/amd64
```

@djencks @squakez Is there a way to make this script work?

**Comment by squakez** [comment]
It's strange. I have it working as usual:
```
$ make generate-doc
./script/gen_doc.sh
Generating API documentation...
Generating CRD API documentation...
I0427 10:41:42.890875   20160 main.go:135] parsing go packages in directory github.com/apache/camel-k/pkg/apis/camel/v1
I0427 10:41:50.145351   20160 main.go:239] using package=github.com/apache/camel-k/pkg/apis/camel/v1
W0427 10:41:50.158867   20160 main.go:461] not found external link source for type encoding/xml.Name
I0427 10:41:50.159911   20160 main.go:175] written to ./script/gen_crd/../../docs/modules/ROOT/partials/apis/camel-k-crds.adoc
I0427 10:41:50.817767   20987 main.go:135] parsing go packages in directory github.com/apache/camel-k/pkg/apis/camel/v1alpha1
I0427 10:41:58.157475   20987 main.go:239] using package=github.com/apache/camel-k/pkg/apis/camel/v1alpha1
W0427 10:41:58.161418   20987 main.go:461] not found external link source for type encoding/json.Number
W0427 10:41:58.161472   20987 main.go:461] not found external link source for type encoding/json.Number
W0427 10:41:58.161633   20987 main.go:461] not found external link source for type encoding/json.Number
I0427 10:41:58.163596   20987 main.go:175] written to ./script/gen_crd/../../docs/modules/ROOT/partials/apis/kamelets-crds.adoc
Generating CRD API documentation... Done.
Generating API documentation... done!
Generating traits documentation...
Generating traits documentation... done!
```
Indeed, I have version 1.7 of Go:
```
$ go version
go version go1.17 linux/amd64
```
Could it be because of this?

**Comment by tadayosi** [comment]
@squakez Yes, you're right. When I switch to Go 1.17.x it runs fine too. I've been using 1.16.x since it's the lowest version we use for Camel K builds.

**Comment by squakez** [comment]
We have a long standing PR open about that: https://github.com/apache/camel-k/pull/2845 as far I can see it implied the update of several kube dependencies. I am not sure if we want to proceed with those upgrades as well. @astefanutti wdyt?

**Comment by tadayosi** [comment]
The last release of K8s which is built on Go 1.16.x is 1.22.x, and its EOL is 2022-10-28.
https://kubernetes.io/releases/#release-v1-22
So at the latest we can safely upgrade to Go 1.17.x after the date. Of course, we can upgrade it earlier if we want.

**Comment by tadayosi** [comment]
Should be fixed once #2845. I go ahead and close this.
